contrib/inventory/freeipa.py

```diff
@@ -3,18 +3,41 @@
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 import argparse
-from ipalib import api
+from distutils.version import LooseVersion
+from ipalib import api, __version__ as IPA_VERSION
 import json
+import os
+import sys
 
 
 def initialize():
     '''
     This function initializes the FreeIPA/IPA API. This function requires
     no arguments. A kerberos key must be present in the users keyring in
-    order for this to work.
+    order for this to work. IPA default configuration directory is /etc/ipa,
+    this path could be overridden with IPA_CONFDIR environment variable.
     '''
 
     api.bootstrap(context='cli')
+
+    if not os.path.isdir(api.env.confdir):
+        print("WARNING: IPA configuration directory (%s) is missing. "
+              "Environment variable IPA_CONFDIR could be used to override "
+              "default path." % api.env.confdir)
+
+    if LooseVersion(IPA_VERSION) == LooseVersion('4.6.2'):
+        # With ipalib < 4.6.0 'server' and 'domain' have default values
+        # ('localhost:8888', 'example.com'), newer versions don't and
+        # DNS autodiscovery is broken, then one of jsonrpc_uri / xmlrpc_uri is
+        # required.
+        # ipalib 4.6.0 is unusable (https://pagure.io/freeipa/issue/7132)
+        # that's why 4.6.2 is explicitely tested.
+        if 'server' not in api.env or 'domain' not in api.env:
+            print("ERROR: ('jsonrpc_uri' or 'xmlrpc_uri') or 'domain' are not "
```

Comment:
print can be replaced using sys.exit 

Except this looks good to me.

Reply:
updated